### PR TITLE
Mixed type API documentation is missing

### PR DESCRIPTION
**Pull request by ihsandemir**
Added documentation for mixed type support implementation. Also a minor API correction/addition to the config.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -314,7 +317,54 @@ After you implement your serializer, you can register it using `SerializationCon
 
 ```
 clientConfig.getSerializationConfig().
-registerSerializer(boost::shared_ptr<hazelcast::client::serialization::SerializerBase>(new MyCustomSerializer());
+registerSerializer(boost::shared_ptr<hazelcast::client::serialization::StreamSerializer>(new MyCustomSerializer());
+```
+
+## Polymorphic Types Serialization
+The client library can handle polymorphic objects such that you can use a base class type structure such as IMap, and put and get derived types. This support exists for IdentifiedDataSerializable, Portable and Custom objects.
```
Comment (Serdaro) [inline README.md]: ...structure, such as IMap, and put....

File: README.md [hunk 2]
```diff
@@ -314,7 +317,54 @@ After you implement your serializer, you can register it using `SerializationCon
...
+
+You can find the polymorphic usage examples at examples/distributed-map/mixed-map folder.
+
+For IdentifiedDataSerializable objects, you need to register a factory implementation of type serialization::DataSerializableFactory. E.g.:
```
Comment (Serdaro) [inline README.md]: ...Factory,. See the below example.

File: README.md [hunk 3]
```diff
@@ -385,6 +435,58 @@ value = vals->get(0);
 
 Using raw pointer based API may improve performance if you are using the API to return multiple values such as values, keySet, and entrySet. In this case, cost of deserialization is delayed until the item is actually accessed.
 
+# Mixed Object Types Supporting HazelcastClient
+Sometimes, you may need to use Hazelcast data structures with objects of different types. E.g. you may want to put int, string,identifieddataserializable, etc... objects into the same Hazelcast IMap data structure. You can do this by using the mixed type adopted HazelcastClient. You can adopt the client in this way:
```
Comment (Serdaro) [inline README.md]: ...of different types. For example, you may want to put `int`, `string`, `identifieddataserializable`, etc. objects...

File: README.md [hunk 4]
```diff
@@ -385,6 +435,58 @@ value = vals->get(0);
...
+```
+    ClientConfig config;
+    HazelcastClient client(config);
+    mixedtype::HazelcastClient &hazelcastClient = client.toMixedType();
+``` 
+The mixedtype::HazelcastClient interface is designed to provide you the data structures which allows you to work with any object types in a mixed manner. E.g. The interface allows you to provide the key and value type differently for each map.put call. An example usage:
```
Comment (Serdaro) [inline README.md]: ...mixed manner. For example, the interface allows......An example usage is shown below:

File: README.md [hunk 5]
```diff
@@ -385,6 +435,58 @@ value = vals->get(0);
...
+```
+    mixedtype::IMap map = hazelcastClient.getMap("MyMap");
+
+    map.put<int, int>(3, 5);
+    map.put<string, std::string>("string key1", "MyStringValue");
+    map.put<int, MyCustomObject>("string key1", myCustomInstance);
+    TypedData result = map.get<int>(3);
+``` 
+
+As you can see in the above code snippet, we are putting int, string and MyCustomObject to the same map. Both the key and value can be different type for each map.put call.
```
Comment (Serdaro) [inline README.md]: ...value can be of different type for.....

File: README.md [hunk 6]
```diff
@@ -385,6 +435,58 @@ value = vals->get(0);
...
+
+If you want to use mixed type map with near cache, then you should use the MixedNearCacheConfig class and add config to the ClientConfig using the addMixedNearCacheConfig method. e.g.:
```
Comment (Serdaro) [inline README.md]: ...Config method. See the below example:

File: README.md [hunk 7]
```diff
@@ -385,6 +435,58 @@ value = vals->get(0);
...
+```
+   boost::shared_ptr<mixedtype::config::MixedNearCacheConfig> nearCacheConfig(new mixedtype::config::MixedNearCacheConfig("MixedMapTestMap"));
+   clientConfig.addMixedNearCacheConfig(nearCacheConfig);
+```
+Mixed type support for near cache only exist when the in-memory format is BINARY. The OBJECT in-memory format is not supported for MixedNearCacheConfig;
```
Comment (Serdaro) [inline README.md]: .....cache only exists when....

File: README.md (hunk 1 above)
Comment (ihsandemir) [reply README.md]: OK

File: README.md (hunk 2 above)
Comment (ihsandemir) [reply README.md]: OK, done

File: README.md (hunk 3 above)
Comment (ihsandemir) [reply README.md]: OK, done.

File: README.md (hunk 4 above)
Comment (ihsandemir) [reply README.md]: Ok, done.

File: README.md (hunk 6 above)
Comment (ihsandemir) [reply README.md]: OK, done.

File: README.md (hunk 7 above)
Comment (ihsandemir) [reply README.md]: Ok, done.

File: README.md (hunk 5 above)
Comment (ihsandemir) [reply README.md]: ok, done.

**Pull request merged**